src/engine/EventManager.cxx

```diff
@@ -352,7 +352,17 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
           // if we touch a bigger than 1x1 tile also add all nodes of the building to highlight.
           for (const auto &coords : m_nodesToHighlight)
           {
-            const Layer layer = TileManager::instance().getTileLayer(tileToPlace);
+            // If we place a ground decoration tile, we must add all tiles of bigger than 1x1 buildings from the Layer BUILDINGS
+            Layer layer;
+            if (TileManager::instance().getTileData(tileToPlace) &&
+                TileManager::instance().getTileData(tileToPlace)->tileType == +TileType::GROUNDDECORATION)
```

Comment:
this check in getTileLayer wouldn't make sense, since we want the actual layer that will be used for the tile there. 
Here we overwrite the layer to BUILDINGS to gather all nodes of a multiobject. I've replaced the check with isNodeMultiObject()now which makes more sense than to check for ground decoration